misc.Rmd

```diff
@@ -65,69 +80,45 @@ citation()
-
-The convention is to put scripts of this nature into a subdirectory of `inst/`, `inst/python`, `inst/perl`, `inst/ruby` etc.
-If these scripts are essential to your package, make sure you also add the appropriate programming language to the `SystemRequirements` field in the `DESCRIPTION`.
-(This field is for human reading so don't worry about exactly how you specify it.)
+You can call `usethis::use_citation()` to initiate this file and fill in your details.
+Read the `?bibentry` help topic for more details.
 
-Java is a special case and the best place to learn more is the documentation of the rJava package (<http://www.rforge.net/rJava/>).
+## Configuration tools
 
-## Demos {#demo}
+If a package has a configuration script (`configure` on Unix-alikes, `configure.win` on Windows), it is executed as the first step by `R CMD  INSTALL`. And if the configuration script needs auxiliary files, those should be located in the `tools/` directory.
+The scripts below `tools/` can have an effect on the installed package, but the contents of `tools/` will not ultimately be present in the installed package.
+In any case, this is mostly (but not solely) relevant to packages containing compiled code, which is beyond the scope of this book.
 
-A demo is an `.R` file that lives in `demo/`.
-Demos are like examples but tend to be longer.
-Instead of focussing on a single function, they show how to weave together multiple functions to solve a problem.
+We bring this up because, in practice, some packages use the `tools/` directory for a different but related purpose.
+Some packages have periodic maintenance tasks for which it is helpful to record detailed instructions.
+For example, many packages embed some sort of external resource, e.g. code or data:
 
-You list and access demos with `demo()`:
+-   Source code and headers for an embedded third party C/C++ library
 
--   Show all available demos: `demo()`.
--   Show all demos in a package: `demo(package = "httr")`.
--   Run a specific demo: `demo("oauth1-twitter", package = "httr")`.
--   Find a demo: `system.file("demo", "oauth1-twitter.R", package = "httr")`.
+-   Web toolkits
 
-Each demo must be listed in `demo/00Index` in the following form: `demo-name   Demo description`.
-The demo name is the name of the file without the extension, e.g. `demo/my-demo.R` becomes `my-demo`.
+-   R code that's inlined (as opposed to imported)
 
-By default the demo asks for human input for each plot: "Hit <Return> to see next plot:".
-This behaviour can be overridden by adding `devAskNewPage(ask = FALSE)` to the demo file.
-You can add pauses by adding: `readline("press any key to continue")`.
+-   Specification for a web API
 
-Generally, I do not recommend using demos.
-Instead, consider writing a vignette:
+-   Colour palettes, styles, and themes
 
--   Demos are not automatically tested by `R CMD check`.
-    This means that they can easily break without your knowledge.
+-   Database of time zones
```

Comment:
tzdb

Reply:
Maybe just say "other external data" since tzdb is the only package that ever needs to include the tz database?

Reply:
Well, I'll just remove this one then. It's a list of examples, so it doesn't make sense to list "other".